src/lib/eina/eina_thread_posix.c

```diff
@@ -0,0 +1,166 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+
+#include "eina_config.h"
+#include "eina_lock.h" /* it will include pthread.h with proper flags */
+#include "eina_thread.h"
+#include "eina_sched.h"
+#include "eina_cpu.h"
+
+/* undefs EINA_ARG_NONULL() so NULL checks are not compiled out! */
+#include "eina_safety_checks.h"
+
+#include "eina_debug_private.h"
+
+#include <pthread.h>
+#include <errno.h>
+
+#include <signal.h>
+
+#if defined(EINA_HAVE_PTHREAD_AFFINITY) || defined(EINA_HAVE_PTHREAD_SETNAME)
+#ifndef __linux__
+#include <pthread_np.h>
+#define cpu_set_t cpuset_t
+#endif
+#endif
+
+inline void *
+_eina_thread_join(Eina_Thread t)
+{
+   void *ret = NULL;
+   int err = pthread_join((pthread_t)t, &ret);
+
+   if (err == 0) return ret;
+   return NULL;
+}
+
+inline Eina_Bool
+_eina_thread_create(Eina_Thread *t, int affinity, void *(*func)(void *data), void *data)
+{
+   int err;
+   pthread_attr_t attr;
+
+   sigset_t oldset, newset;
+
+   pthread_attr_init(&attr);
+   if (affinity >= 0)
+     {
+#ifdef EINA_HAVE_PTHREAD_AFFINITY
+      cpu_set_t cpu;
+
+      CPU_ZERO(&cpu);
+      CPU_SET(affinity, &cpu);
+      pthread_attr_setaffinity_np(&attr, sizeof(cpu), &cpu);
+#endif
+     }
+
+   /* setup initial locks */
+
+   sigemptyset(&newset);
+   sigaddset(&newset, SIGPIPE);
+   sigaddset(&newset, SIGALRM);
+   sigaddset(&newset, SIGCHLD);
+   sigaddset(&newset, SIGUSR1);
+   sigaddset(&newset, SIGUSR2);
+   sigaddset(&newset, SIGHUP);
+   sigaddset(&newset, SIGQUIT);
+   sigaddset(&newset, SIGINT);
+   sigaddset(&newset, SIGTERM);
+#ifdef SIGPWR
+   sigaddset(&newset, SIGPWR);
+#endif
+   pthread_sigmask(SIG_BLOCK, &newset, &oldset);
+
+   err = pthread_create((pthread_t *)t, &attr, func, data);
+
+   pthread_sigmask(SIG_SETMASK, &oldset, NULL);
+
+   pthread_attr_destroy(&attr);
+
+   if (err == 0) return EINA_TRUE;
+
+   return EINA_FALSE;
+}
+
+inline Eina_Bool
+_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
+{
+   return pthread_equal((pthread_t)t1, (pthread_t)t2);
+}
+
+Eina_Bool _eina_thread_cancel(Eina_Thread thread)
+{
+   return pthread_cancel((pthread_t)thread) == 0;
+}
+
+inline Eina_Thread
+_eina_thread_self(void)
+{
+   return (Eina_Thread)pthread_self();
+}
+inline void _eina_sched_prio_drop(void)
```

Comment:
why an eina_sched.h function is being implemented at einta_thread and not on an eina_sched_{win32,posix}.c file?

Reply:
now it is